File: Example/ExampleObjectiveC/PIOMapViewController.m
//
//  MapViewController.m
//  ExampleObjectiveC
//
//  Created by zee-pk on 31/08/2016.
//  Copyright © 2016 predict.io by ParkTAG GmbH. All rights reserved.
//

#import "PIOMapViewController.h"
#import <MapKit/MapKit.h>

@interface PIOMapViewController ()

@property (weak, nonatomic) IBOutlet MKMapView *mapView;

@end

@implementation PIOMapViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.navigationItem.title = @"Location";
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];

    CLLocationCoordinate2D coordinate = self.location.coordinate;

    MKPointAnnotation *annotation = [[MKPointAnnotation alloc] init];
    [annotation setCoordinate:coordinate];
    [annotation setTitle:@"Event location"]; //You can set the subtitle too
    [self.mapView addAnnotation:annotation];

    MKCircle *circle = [MKCircle circleWithCenterCoordinate:coordinate radius:self.location.horizontalAccuracy];
    [self.mapView addOverlay:circle];

    CLLocationDegrees coordinateDelta = 0.003;
    MKCoordinateSpan span = MKCoordinateSpanMake(coordinateDelta, coordinateDelta);
    MKCoordinateRegion region = MKCoordinateRegionMake(coordinate, span);
    [self.mapView setRegion:region animated:YES];
}

-(MKOverlayRenderer *)mapView:(MKMapView *)mapView rendererForOverlay:(id<MKOverlay>)overlay {
    if ([overlay isKindOfClass:[MKCircle class]]){
        MKCircleRenderer *circleRenderer = [[MKCircleRenderer alloc] initWithCircle:overlay];
        circleRenderer.fillColor = [[UIColor greenColor] colorWithAlphaComponent:0.05];
        circleRenderer.strokeColor = [[UIColor greenColor] colorWithAlphaComponent:0.3];
        circleRenderer.lineWidth = 4;
        return circleRenderer;
    }
    return nil;
}

- (MKAnnotationView *)mapView:(MKMapView *)mapView viewForAnnotation:(id <MKAnnotation>)annotation {
    MKAnnotationView *annotationView = nil;
    if (![annotation isKindOfClass:[MKUserLocation class]]) {
        annotationView = [[MKPinAnnotationView alloc] initWithAnnotation:annotation reuseIdentifier:@"temp"];
        annotationView.canShowCallout = YES;
    }

    return annotationView;
}

@end

File: ExampleSwift/ExampleSwift/PIOMapViewController.swift
//
//  MapViewController.swift
//  ExampleSwift
//
//  Created by zee-pk on 08/09/2016.
//  Copyright (c) 2016 predict.io by ParkTAG GmbH. All rights reserved.
//

import UIKit
import CoreData
import Foundation
import MapKit
import PredictIO

class PIOMapViewController: UIViewController, MKMapViewDelegate {
    
    var location: CLLocation!
    var zoneType = PIOZoneType.other
    
    @IBOutlet weak var mapView: MKMapView!
    @IBOutlet weak var zoneLabel: UILabel!
    @IBOutlet weak var zoneView: UIView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.navigationItem.title = "Location"
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        let coordinate = self.location.coordinate
        let annotation = MKPointAnnotation()
        annotation.coordinate = coordinate
        annotation.title = "Event location" //You can set the subtitle too
        mapView.addAnnotation(annotation)
        
        mapView.add(MKCircle(center: coordinate, radius: self.location.horizontalAccuracy))
        
        let coordinateDelta:CLLocationDegrees = 0.003
        let span = MKCoordinateSpanMake(coordinateDelta, coordinateDelta)
        let region = MKCoordinateRegionMake(coordinate, span)
        mapView.setRegion(region, animated: true)
        
        // display zone information if available
        switch zoneType {
        case .other:
            zoneView.isHidden = true
            break
        case .home:
            zoneLabel.text = "Event triggered within Home zone"
            zoneView.isHidden = false
            break
        case .work:
            zoneLabel.text = "Event triggered within Work zone"
            zoneView.isHidden = false
            break
        }
    }
    
    func mapView(_ mapView: MKMapView, rendererFor overlay: MKOverlay) -> MKOverlayRenderer {
        if (overlay is MKCircle) {
            let circleRenderer = MKCircleRenderer.init(circle: overlay as! MKCircle)
            circleRenderer.fillColor = UIColor.green.withAlphaComponent(0.05)
            circleRenderer.strokeColor = UIColor.green.withAlphaComponent(0.3)
            circleRenderer.lineWidth = 4
            return circleRenderer
        }
        return MKPolylineRenderer()
    }
    
    func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
        var annotationView: MKAnnotationView? = nil
        if !(annotation is MKUserLocation) {
            annotationView = MKPinAnnotationView(annotation: annotation, reuseIdentifier: "temp")
            annotationView!.canShowCallout = true
        }
        return annotationView!
    }
}
